sync/src/block/extension.rs

```diff
@@ -476,22 +476,27 @@ impl NetworkExtension<Event> for Extension {
         cinfo!(SYNC, "New peer detected #{}", id);
         self.send_status(id);
 
-        let t = self.connected_nodes.insert(*id);
-        debug_assert!(t, "{} is already added to peer list", id);
+        if !self.connected_nodes.contains(id) {
```

Comment:
It seems this patch is avoiding the error case. Not fixing the error.
When the `on_node_added` is called, it should not be included in `connected_nodes`.
Could you try to fix this problem fundamentally?